### PR TITLE
Tests coverage with Codecov ☂️

### PR DESCRIPTION
**Pull request by pedrofurtado**
Hi again, @shirts ! 😄 

I've created the skeleton for setup of Codecov in repo, to provide us the percentage of tests coverage. We can later add some badges in README.md to facilitate the view of this kind of information (for github action CI and coverage for example).

What do you think?

For this PR works well, we just have to do the following: creating an project in Codecov, related to repo, and then get the "secret" generated by codecov and register in Github repo settings as a "Repo secret" with name `CODECOV_TOKEN`. When you have time, could you do this for us (because this can only be done with administrator access)? Once made this action, the coverage will work for future commits/pull requests. We can test it in this PR, anyway, before merge 🎉

### CODE REVIEW

**Comment by shirts** [comment]
`CODECOV_TOKEN` secret added

**Comment by pedrofurtado** [comment]
In next PRs, we can add some badges in README.md to indicate codecov and github CI status. But first we need them merged and running 🥇

**Comment by pedrofurtado** [comment]
@shirts 🤝 🤓

**Comment by pedrofurtado** [comment]
@shirts Are we ready for merge? ✅

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
 source "https://rubygems.org"
 
 gemspec
+gem "codecov", require: false, group: :test
```
Comment (shirts) [inline Gemfile:4]: I'd like to keep all dependencies in the gemspec. This would be a development dependency. Do you have any opinions here?

Comment (pedrofurtado) [reply Gemfile:4]: Good idea, I will move it to gemspec file 👍

Comment (pedrofurtado) [reply Gemfile]: Moved ✅

**Review by shirts — CHANGES_REQUESTED** [review]
Squash commits 🙏

**Pull request closed without merging**